c2rust-analyze/src/main.rs

```diff
@@ -278,6 +278,35 @@ fn update_pointer_info<'tcx>(acx: &mut AnalysisCtxt<'_, 'tcx>, mir: &Body<'tcx>)
     }
 }
 
+// Walks the type `ty` and applies a function `f` to it if it's an ADT
+// `f` gets applied recursively to `ty`'s generic types and fields (if applicable)
+fn walk_args_and_fields<'tcx, F>(tcx: TyCtxt<'tcx>, ty: Ty<'tcx>, f: &mut F)
```

Comment:
I think `walk_adts` would be a better name.  `f` is called on all ADTs found.

Reply:
Now that this is also walking a `FnPtr` I've renamed to `walk_ffi_safe_ty` in 88472f72

Reply:
29fa2691